docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/oci-download.md

```diff
@@ -0,0 +1,144 @@
+---
+sidebar_label: oci-download
+description: Downloads OCI artifacts from a registry and extracts layer content to a specified file.
+---
+
+# `oci-download`
+
+`oci-download` downloads OCI artifacts from a registry and extracts layer
+content to a specified file. This step is useful for downloading artifacts like
+Helm charts, configuration files, or other resources packaged as OCI artifacts.
+The step supports authentication and can target specific layers by media type.
+
+:::note
+Downloads are limited to 100MB to prevent resource exhaustion.
+:::
+
+## Configuration
+
+| Name | Type | Required | Description |
+|------|------|----------|-------------|
+| `imageRef` | `string` | Y | Reference to the OCI artifact to download. Supports both tag format `registry/repository:tag` and digest format `registry/repository@sha256:digest`. For Helm OCI artifacts, the `oci://` prefix is supported (e.g., `oci://registry/repository:tag`) and will use Helm-specific credential lookup. |
+| `outPath` | `string` | Y | Path to the destination file where the extracted artifact will be saved. This path is relative to the temporary workspace that Kargo provisions for use by the promotion process. |
+| `allowOverwrite` | `boolean` | N | Whether to allow overwriting an existing file at the specified path. If `false` and the file exists, the download will fail. Defaults to `false`. |
+| `mediaType` | `string` | N | Media type of the layer to download. Selects the first layer matching this type. If not specified, selects the first layer available. |
+| `insecureSkipTLSVerify` | `boolean` | N | Whether to skip TLS verification when downloading the artifact. Defaults to `false`. |
+
+## Examples
+
+### Downloading a Helm Chart
+
+In this example, an OCI-packaged Helm chart is downloaded from a registry and
+saved to a local file. This is useful when you need to download charts that are
+distributed as OCI artifacts.
+
+```yaml
+steps:
+- uses: oci-download
+  config:
+    imageRef: registry.example.com/charts/my-app:1.0.0
+    outPath: ./charts/my-app-1.0.0.tgz
+    mediaType: application/vnd.cncf.helm.chart.content.v1.tar+gzip
```

Comment:
The "Downloading a Helm Chart" examples showing mediaType is why I had concerns. People will copy these examples, and so we'll be promoting the use of this field unintentionally.

Reply:
I have updated the examples.